### DISCUSSION TITLE
Web3Modal wallet provider sign message vanilla JS

### DISCUSSION THREAD

**Discussion by juloxrox** [WalletConnect/walletconnect-monorepo#4390] [Developer Support] [answered]
Hi everyone,

I got the following working code to allow users to connect to my Solana dapp with web3modal (with Solflare or Phantom):

```
import { createWeb3Modal, defaultSolanaConfig } from '@web3modal/solana'
const projectId = import.meta.env.VITE_WALLETCONNECT_PROJECT_ID

// 2. Create wagmiConfig
const metadata = {
    name: import.meta.env.VITE_APP_NAME,
    description: '',
    url: import.meta.env.VITE_APP_URL
    icons: []
}

const solana = {
    chainId: '5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp',
    name: 'Solana',
    currency: 'SOL',
    explorerUrl: 'https://solscan.io',
    rpcUrl: 'https://rpc.walletconnect.com/v1'
}
const chains = [solana]
export const config = defaultSolanaConfig({
    chains,
    projectId,
    metadata,
})

// 3. Create modal
const modal = createWeb3Modal({
    solanaConfig: config,
    projectId,
    metadata,
    chains,
    enableAnalytics: true, // Optional - defaults to your Cloud configuration
    enableOnramp: false, // Optional - false as default
    defaultChain: solana
})
```

Now I would like users to sign a message, and I found the following snippet but I'm missing something:

```
async signMessage() {
        try {
            const encodedMessage = new TextEncoder().encode('Hello from Web3Modal')
            const signature = await walletProvider.signMessage(encodedMessage)

            if (signature.signature) {
                console.log(signature.signature)
            }
        } catch (e) {
            console.error(e)
        }
}
```

How can I define the "**walletProvider**" object? (dependencies + code please)

Thanks in advance!

**Comment by enesozturk** [comment, marked as answer]
Hey @juloxrox, thanks for reaching out. You can access the same exported members by using the methods in `modal` object which is created with the `createWeb3Modal` function.

```js
const modal = createWeb3Modal({ ... })
...
const walletProvider = modal.getWalletProvider()
```

Also, I'd like to tell you that the Solana client is not launched yet and you might expect some issues. We are planning to publish it this week. So if you encounter any issues, please share with us and we are happy to help.

**Reply by juloxrox** [reply]
Hello @enesozturk,

Alright yes I tried this method but this was only working on evm client but then it's all good for me I will just wait for the release of Solana client this week and update dependencies.

Thanks for your support!

**Reply by enesozturk** [reply]
@juloxrox, do you mean that it's not working with the current Solana client even though you use it this way?
